bin/generator-utils/templates.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+# shellcheck source=/dev/null
+source ./bin/generator-utils/utils.sh
+
+function create_test_file_template() {
+    local exercise_dir=$1
+    local slug=$2
+    local test_file="${exercise_dir}/tests/${slug}.rs"
+
+    cat <<EOT >"$test_file"
+use $(dash_to_underscore "$slug")::*;
+// Add tests here
+
+EOT
+
+    # fetch canonical_data
+    canonical_json=$(curl https://raw.githubusercontent.com/exercism/problem-specifications/main/exercises/"$slug"/canonical-data.json)
+
+    if [ "${canonical_json}" == "404: Not Found" ]; then
+        canonical_json=$(jq --null-input '{cases: []}')
+
+        cat <<EOT >>"$test_file"
+// As there isn't a canonical data file for this exercise, you will need to craft your own tests.
+// If you happen to devise some outstanding tests, do contemplate sharing them with the community by contributing to this repository:
+// https://github.com/exercism/problem-specifications/tree/main/exercises/${slug}
+EOT
+    fi
+
+    # sometimes canonical data has multiple levels with multiple `cases` arrays.
+    #(see kindergarten-garden https://github.com/exercism/problem-specifications/blob/main/exercises/kindergarten-garden/canonical-data.json)
+    # so let's flatten it
+    cases=$(echo "$canonical_json" | jq '[ .. | objects | with_entries(select(.key | IN("uuid", "description", "input", "expected"))) | select(. != {}) | select(has("uuid")) ]')
+
+    first_iteration=true
+    # loop through each object
+    jq -c '.[]' <<<"$cases" | while read -r case; do
+        desc=$(echo "$case" | jq '.description' | tr '[:upper:]' '[:lower:]' | tr ' ' '_' | tr -cd '[:alnum:]_' | sed 's/^/test_/')
+        input=$(echo "$case" | jq '.input')
+        expected=$(echo "$case" | jq '.expected')
+
+        # append each test fn to the test file
+        cat <<EOT >>"$test_file"
+#[test] $([[ "$first_iteration" == false ]] && printf "\n#[ignore]")
+fn ${desc}(){
```

Comment:
```suggestion
fn ${desc}() {
```

Reply:
Maybe run `cargo fmt` in the exercise directory to ensure everything is formatted properly?